docs/moving-an-app-off-static.md

```diff
@@ -0,0 +1,124 @@
+# Moving an app off Static
+
+As part of the effort to retire static, the frontend apps that use it will have to be
+altered so that they no longer use the slimmer gem. This means that they will have to
+be responsible for their own Global and Emergency banners, and for configuring the
+layout_for_public component which does the actual laying out of GOV.UK outward-facing
+pages. To move an app off static, you will need to complete the following steps:
+
+## 1. Remove Slimmer gem and config
+
+[example commit](https://github.com/alphagov/email-alert-frontend/pull/1890/commits/6273b19abdd85432058a5417c9fcb32c6900b5b9)
+
+First, remove the gem reference and rebuild the lockfile. Then search for references
+to slimmer. There will be at least one, in the application controller where slimmer
+is included, and possibly in other controllers if they need to set slimmer_template.
+
+Make a note of controllers that set a different slimmer template. They may indicate a
+need either for a specialist layout, or for config variables to send to the default
+application layout.
+
+You may also need to remove slimmer references in test setups.
+
+## 2. Update the application layout to call the layout_for_public component directly
+
+[example commit](https://github.com/alphagov/email-alert-frontend/pull/1890/commits/f80d16bf9dd44093c5541b5802bafe2d41eea7bb)
+
+With static, app layouts are an HTML page, with a body that contains a wrapper element.
+This element is replaced by slimmer with the contents of the relevant template from
+static (which is in turn the body section of a layout_for_public component).
+
+Since layout_for_public is itself a full HTML page, you replace the entire section
+with a call to the layout_for_public component. You will need to add a `content_for :head`
+block. To this, add the gem component stylesheets for the components used in
+layout_for_public, and the call to `render_component_stylesheets`, and also
+any additional head elements that were in the existing layout.
+
+Since the app is now not getting the default styles from static, configure the
+publishing components gem to not exclude static css files.
+
+in `config/initializers/govuk_publishing_components.rb`:
+```
+GovukPublishingComponents.configure do |c|
+  c.component_guide_title = "Email Alert Frontend Component Guide"
+  c.application_stylesheet = "application"
+  c.exclude_css_from_static = false
```

Comment:
Unrelated question: presumably at some point in the future we can remove this entirely, as there'll be no code anywhere that even attempts to get CSS from static?

Reply:
Yep, I've added a note to that effect.